test/ncengine/graphics2/CMakeLists.txt

```diff
@@ -224,6 +224,7 @@ add_executable(PassManifest_tests
 PassManifest_tests.cpp
     ${NC_SOURCE_DIR}/graphics2/GraphicsUtility.cpp
     ${NC_SOURCE_DIR}/graphics2/PostProcess.cpp
+    ${NC_SOURCE_DIR}/graphics2/diligent/ShaderFactory.cpp
```

Comment:
the diligent directory should really only appear in the integration tests. Can we stub whatever was required here?

Reply:
It relied on too many symbols from Diligent, so I've moved it to integration_tests